source/extensions/filters/udp/udp_proxy/hash_policy_impl.cc

```diff
@@ -37,24 +39,28 @@ class KeyHashMethod : public HashPolicyImpl::HashMethod {
   const uint64_t hash_;
 };
 
-HashPolicyImpl::HashPolicyImpl(
-    const absl::Span<const UdpProxyConfig::HashPolicy* const>& hash_policies) {
+absl::optional<uint64_t>
+HashPolicyImpl::generateHash(const Network::Address::Instance& downstream_addr) const {
+  return hash_impl_->evaluate(downstream_addr);
+}
+
+HashPolicyImplPtr HashPolicyImplFactory::create(
+    const Protobuf::RepeatedPtrField<
+        envoy::extensions::filters::udp::udp_proxy::v3::UdpProxyConfig_HashPolicy>& hash_policies) {
   ASSERT(hash_policies.size() == 1);
-  switch (hash_policies[0]->policy_specifier_case()) {
+  switch (hash_policies[0].policy_specifier_case()) {
   case UdpProxyConfig::HashPolicy::PolicySpecifierCase::kSourceIp:
-    hash_impl_ = std::make_unique<SourceIpHashMethod>();
+    return std::make_unique<HashPolicyImpl>(HashPolicyImpl(std::make_unique<SourceIpHashMethod>()));
```

Comment:
a bit confused why you're  creating the HashPolicyImpl twice? once via the `HashPolicyImpl(HashMethodPtr hash_impl)` ctor and another via the unique_ptr? Here and below